### ISSUE TITLE
Automatically restart when a version mismatch is detected

### ISSUE THREAD

**Issue by bronson** [rails/spring#471]
Seems like a lot of people are getting bit by this message: https://github.com/rails/spring/issues/295

```
There is a version mismatch between the spring client and the server.
You should restart the server and make sure to use the same version.

CLIENT: 1.6.3, SERVER: 1.6.2
```

At the very least, the message could be a LOT more helpful.  Something like:

```
Spring 1.6.2 is running but 1.6.3 is needed.  Please restart by running:

    spring stop
```

BUT, why does Spring need user intervention here?  When it detects a version mismatch, it could print a warning, restart the server, and get on with life.
```
Stopped the running Spring 1.6.2 server, launching 1.6.3.
```
No need to hassle humans with an easy-to-fix problem!

If either of these fixes sound OK, I can bang together a PR.  Personally, I prefer the latter.

**Comment by arktisklada** [comment]
This was resolved by #476 
Can you confirm @bronson ?

**Comment by bronson** [comment]
That looks like a great solution, nicely done @arktisklada.  I won't be able to try it for a few days...  Since it probably works as advertised, I'll close this issue and re-open if I run into issues.
